appserver/tests/test-micro-jar/pom.xml

```diff
@@ -0,0 +1,87 @@
+    file that accompanied this code.
+
+    Modifications:
+    If applicable, add the following below the License Header, with the fields
+    enclosed by brackets [] replaced by your own identifying information:
+    "Portions Copyright [year] [name of copyright owner]"
+
+    Contributor(s):
+    If you wish your version of this file to be governed by only the CDDL or
+    only the GPL Version 2, indicate your decision by adding "[Contributor]
+    elects to include this software in this distribution under the [CDDL or GPL
+    Version 2] license."  If you don't indicate a single choice of license, a
+    recipient has the option to distribute your version of this file under
+    either the CDDL, the GPL Version 2 or to extend the choice of license to
+    its licensees as provided above.  However, if you add GPL Version 2 code
+    and therefore, elected the GPL Version 2 license, then the option applies
+    only if the new code is made subject to such option by the copyright
+    holder.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.glassfish.main.tests</groupId>
+        <artifactId>tests</artifactId>
+        <version>5.191-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>test-micro-jar</artifactId>
+
+    <name>Integration Tests based on payara-micro.jar file</name>
+    <description>NOTE: this module is dependent on the payara-micro-distribution being run previously 
+    but it cannot have a dependency on this module as this would mess up the classpath for test execution 
+    as some classes get in from the modules that have to be taken only from the generated jar file.
+
+    The idea is to test the actual artifact created.
+    This is not suitable for all kinds of tests as the accessible API is the minumal payara micro API.
+    </description>
+
+    <dependencies>
+        <!-- NOTE that the dependency is of type "pom". We don't want its jars in classpath - 
+        only that the module is build before this one so that the payara-micro.jar exists -->
+        <dependency>
+            <groupId>fish.payara.extras</groupId>
+            <artifactId>payara-micro</artifactId>
+            <version>5.191-SNAPSHOT</version>
+            <type>pom</type>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <!-- NOTE: This is a purely fictional module. Maven simply insists that anything is a module 
+            so we pretend it is -->
+            <groupId>fish.payara.extras</groupId>
```

Comment:
Short answer: Maven. 

Long answer: Only way to make maven accept "external jars" and to get them on the compile and runtime classpath is to pretend they are an artifact of type `system`. It basically means maven picks up the jar in the given path and assumes it is the artifact with the given group and artifact ID and version but non of the three values given does actually matter or have any function (as long as they don't collide with existing ones). Since the tags are mandatory you just have to provide something.

Reply:
Nice, love the workaround in that case!